openmc/material.py

```diff
@@ -311,6 +327,65 @@ def remove_nuclide(self, nuclide):
         if nuclide._name in self._nuclides:
             del self._nuclides[nuclide._name]
 
+    def add_macroscopic(self, macroscopic):
+        """Add a macroscopic to the material
+
+        Parameters
+        ----------
+        macroscopic : str or openmc.macroscopic.Macroscopic
+            Macroscopic to add
+
+        """
+
+        # Ensure no nuclides, elements, or sab are added since these would be
+        # incompatible with macroscopics
+        if ((len(self._nuclides.keys()) != 0) and
+            (len(self._elements.keys()) != 0) and (len(self._sab) != 0)):
+            msg = 'Unable to add a Macroscopic data set to Material ID="{0}" ' \
+                  'with a macroscopic value "{1}" as an incompatible data ' \
+                  'member (i.e., nuclide, element, or S(a,b) table) ' \
+                  'has already been added'.format(self._id, macroscopic)
+            raise ValueError(msg)
+
+        if not isinstance(macroscopic, (openmc.Macroscopic, str)):
+            msg = 'Unable to add a Macroscopic to Material ID="{0}" with a ' \
+                  'non-Macroscopic value "{1}"'.format(self._id, macroscopic)
+            raise ValueError(msg)
+
+        if isinstance(macroscopic, openmc.Macroscopic):
+            # Copy this Macroscopic to separate it from the Macroscopic in
+            # other Materials
+            macroscopic = deepcopy(macroscopic)
+        else:
+            macroscopic = openmc.Macroscopic(macroscopic)
+
+        if self._macroscopic is None:
```

Comment:
If you switch to using a `macroscopic` property setter instead, there would be no need to report an error msg if the `Material` already has a `Macroscopic` (one could simply interchange `Macroscopics` in between runs from within Python)